javagen/src/main/java/com/azure/autorest/mapper/ClientMapper.java

```diff
@@ -591,7 +594,11 @@ private static ModuleInfo getModuleInfo(List<String> modelsPackages) {
             }
 
             // open models package to azure-core and jaskson
-            List<String> openToModules = Arrays.asList("com.azure.core", "com.fasterxml.jackson.databind");
+            List<String> openToModules = new ArrayList<>();
+            openToModules.add(Package.CORE.getPackageName());
+            if (!settings.isStreamStyleSerialization()) {
```

Comment:
Do we have any documents on StreamStyleSerialization? I found I'm lack of knowledge of StreamStyleSerialization, and want to learn mot. Sorry it may not really related to this pr.

Reply:
Not sure if we had good doc on it. You can ping Alan in chat.

It mostly about model. It would use `JsonSerializable` and the class do serialization, not Jackson via reflect.
https://github.com/Azure/autorest.java/tree/main/vanilla-tests/src/main/java/fixtures/streamstyleserialization/models

It actually have tight relation to your Patch work.